### PR TITLE
MIGENG-407 populate new fields for cnv hotplug cpu/memory flag

### PR DESCRIPTION
**Pull request by m-brophy**
https://issues.redhat.com/browse/MIGENG-407

End2EndTest will fail until 

https://github.com/project-xavier/xavier-analytics/pull/54 is merged into master of xavier-analytics

### CODE REVIEW

File: src/test/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculatorTest.java [hunk 1]
```diff
@@ -113,6 +116,7 @@ public void calculate_jsonV1_0_0_Given_ShouldReturnCalculatedValues() throws IOE
         expectedModel.setVersion("6.7.2");
         expectedModel.setProduct("VMware vCenter");
         expectedModel.setScanRunDate(new SimpleDateFormat("yyyy-M-dd'T'hh:mm:ss.S").parse("2019-09-18T14:52:45.871Z"));
+        expectedModel.setHasCpuHotRemove(false);
```
Comment (jonathanvila) [inline src/test/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculatorTest.java:120]: You are only setting expectancy for 1 of the 3 new filds... any reason for that ?

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -124,6 +127,9 @@ cloudforms.manifest.0.vmworkloadinventory.numCpuPath=$.ManageIQ::Providers::Vmwa
 # TODO fix this JSONPath
 cloudforms.manifest.0.vmworkloadinventory.numCoresPerSocketPath=$.ManageIQ::Providers::Vmware::InfraManager.hosts[*].vms[?(@.id == {id})].num_cpu
 cloudforms.manifest.0.vmworkloadinventory.hasRDMDiskPath=$.ManageIQ::Providers::Vmware::InfraManager.hosts[*].vms[?(@.id == {id})].has_rdm_disk
+cloudforms.manifest.0.vmworkloadinventory.memoryHotAddEnabledPath=$.ManageIQ::Providers::Vmware::InfraManager.hosts[*].vms[?(@.id == {id})].memory_hot_add_enabled
```
Comment (jonathanvila) [inline src/main/resources/application.properties]: In version 0 there was no such fields , ¿?

File: src/test/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculatorTest.java [hunk 3]
```diff
@@ -93,6 +93,9 @@ public void calculate_jsonV1_0_0_Given_ShouldReturnCalculatedValues() throws IOE
         assertThat(modelList.stream().filter(e -> e.getGuestOSFullName().equalsIgnoreCase("CentOS 7 (64-bit)")).count()).isEqualTo(1);
         assertThat(modelList.stream().filter(e -> e.getGuestOSFullName().equalsIgnoreCase("Red Hat Enterprise Linux Server release 7.6 (Maipo)")).count()).isEqualTo(6);
         assertThat(modelList.stream().filter(e -> e.getDiskSpace() == (17980588032L)).count()).isEqualTo(1);
+        assertThat(modelList.stream().filter(e -> e.getHasCpuHotAdd() != null).count()).isEqualTo(1);
+        assertThat(modelList.stream().filter(e -> e.getHasCpuHotRemove() != null).count()).isEqualTo(1);
+        assertThat(modelList.stream().filter(e -> e.getHasMemoryHotAdd() != null).count()).isEqualTo(1);
```
Comment (jonathanvila) [inline src/test/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculatorTest.java]: what do you think about testing that a concrete VM has the 3 fields != null and another one having all as null because you dont set any of the 3 fields in the VM ? just a suggestion.... :)

File: src/test/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculatorTest.java (hunk 1 above)
Comment (m-brophy) [reply src/test/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculatorTest.java:120]: Each of the fields is set to trigger the flag in a different VM in the test payload, and the expectedValue test only looks at one of these VMs.

Comment (m-brophy) [reply src/test/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculatorTest.java:120]: I've changed it so they are all set in that VM

File: src/main/resources/application.properties (hunk 2 above)
Comment (m-brophy) [reply src/main/resources/application.properties]: I've removed this and altered code to suit

File: src/test/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculatorTest.java (hunk 3 above)
Comment (m-brophy) [reply src/test/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculatorTest.java]: I've added an extra expectedValue tests which tests one of the VMs with all these fields unset, and altered the other one so all three != null

**Comment by m-brophy** [comment]
All your comments should be addressed now and ready to review again @jonathanvila

File: src/main/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculator.java [hunk 4]
```diff
@@ -131,11 +134,40 @@ private Long getDiskSpaceList(Map vmStructMap) {
...
+
+    private Boolean getMemoryHotAdd(Map vmStructMap) {
+        // If the VM.memory_hot_add_enabled is present use it, if not set value to null
+
+        Object memoryHotAddPathObject = getValueForExpandedPathAndHandlePathNotPresent(MEMORYHOTADDENABLEDPATH, vmStructMap,
+                "Setting value to null.");
+
+        return memoryHotAddPathObject != null ? (Boolean) memoryHotAddPathObject : null;
+    }
+
```
Comment (carlosthe19916) [inline src/main/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculator.java]: @m-brophy The content of `getCpuHotAdd, getCpuHotRemove, getMemoryHotAdd, and getCpuAffinity` are exactly the same, do you think it is worth having a single method and pass the PATH variable as a parameter?

File: src/test/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculatorTest.java [hunk 5]
```diff
@@ -124,6 +130,36 @@ public void calculate_jsonV1_0_0_Given_ShouldReturnCalculatedValues() throws IOE
...
+        expectedModel2.setOsProductName("Linux");
+        expectedModel2.setNicsCount(1);
+        expectedModel2.setMemory(2147483648L);
+        expectedModel2.setHasRdmDisk(false);
+        expectedModel2.setGuestOSFullName("Red Hat Enterprise Linux Server release 7.6 (Maipo)");
+        expectedModel2.setDiskSpace(4833341440L);
+        expectedModel2.setDatacenter("JON TEST DC");
+        expectedModel2.setCpuCores(1);
+        expectedModel2.setCluster("VMCluster");
+        expectedModel2.setSystemServicesNames(Arrays.asList("NetworkManager-dispatcher","NetworkManager-wait-online","NetworkManager"));
```
Comment (carlosthe19916) [inline src/test/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculatorTest.java]:
```suggestion
        expectedModel2.setSystemServicesNames(Arrays.asList("NetworkManager-dispatcher", "NetworkManager-wait-online", "NetworkManager"));
```

**Review by carlosthe19916 — COMMENTED** [review]
@m-brophy I've added just a single comment and a couple of format suggestions.

File: src/main/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculator.java (hunk 4 above)
Comment (m-brophy) [reply src/main/java/org/jboss/xavier/integrations/migrationanalytics/business/VMWorkloadInventoryCalculator.java]: Done

**Review by carlosthe19916 — APPROVED** [review]

**Comment by carlosthe19916** [comment]
@m-brophy Thanks for your work in this PR
